report-ng/app/src/components/timings/session-timings/session-timings.ts

```diff
@@ -0,0 +1,228 @@
+/*
+ * Testerra
+ *
+ * (C) 2023, Telekom MMS GmbH, Deutsche Telekom AG
+ *
+ * Deutsche Telekom AG and all other contributors /
+ * copyright owners license this file to you under the Apache
+ * License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {autoinject} from 'aurelia-framework';
+import {NavigationInstruction, RouteConfig} from "aurelia-router";
+import {AbstractViewModel} from "../../abstract-view-model";
+import * as echarts from 'echarts';
+import {EChartsOption} from 'echarts';
+import "./session-timings.scss";
+import {MethodDetails, StatisticsGenerator} from "services/statistics-generator";
+import {StatusConverter} from "../../../services/status-converter";
+import moment from "moment";
+import {MethodType} from "../../../services/report-model/framework_pb";
+
+@autoinject()
+export class SessionTimings extends AbstractViewModel {
+    private _chart: echarts.ECharts;
+    private _option: EChartsOption;
+    private _sessionMetrics;
```

Comment:
These members could use types. I.e. `_executionStatistics` is used both as an Array and as an `ExecutionStatistics`.